File: src/main/java/com/liyongquan/weeklycontest/bwc50/MinOperations.java
package com.liyongquan.weeklycontest.bwc50;

public class MinOperations {
    public int minOperations(int[] nums) {
        int len = nums.length;
        int cnt = 0;
        for (int i = 1; i < len; i++) {
            if (nums[i] <= nums[i - 1]) {
                cnt += nums[i - 1] - nums[i] + 1;
                nums[i] = nums[i - 1] + 1;
            }
        }
        return cnt;
    }
}

File: algorithm/golang/weekcontest/bwc114/minOperations.go
package bwc114

func minOperations(nums []int, k int) int {
	n := len(nums)
	counter := make([]int, n+1)
	p := n - 1
	cnt := 0
	for cnt < k {
		counter[nums[p]]++
		if counter[nums[p]] == 1 && nums[p] <= k {
			cnt++
		}
		p--
	}
	return n - p - 1
}
